### PR TITLE
deps: updates TinyGo to v0.28.1

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Performance is across the board lower, but the absolute numbers are unlikely concerning to most users

```
goos: darwin
goarch: arm64
pkg: github.com/http-wasm/http-wasm-guest-tinygo/e2e
                                        │ v0.27.0.txt │             v0.28.1.txt             │
                                        │   sec/op    │    sec/op     vs base               │
/example_router_host_response/TinyGo-12   612.7n ± 0%   668.2n ±  1%   +9.06% (p=0.002 n=6)
/get_uri/TinyGo-12                        617.0n ± 2%   636.9n ±  1%   +3.23% (p=0.002 n=6)
/get_header_values_not_exists/TinyGo-12   506.9n ± 1%   558.5n ±  1%  +10.18% (p=0.002 n=6)
/example_router_wasm_response/TinyGo-12   700.1n ± 1%   818.3n ±  3%  +16.89% (p=0.002 n=6)
/log/TinyGo-12                            448.6n ± 1%   483.1n ±  1%   +7.70% (p=0.002 n=6)
/get_header_names_none/TinyGo-12          694.1n ± 4%   684.4n ±  1%        ~ (p=0.065 n=6)
/get_header_names/TinyGo-12               672.5n ± 0%   687.3n ±  2%   +2.20% (p=0.002 n=6)
/remove_header_exists/TinyGo-12           518.0n ± 1%   549.1n ±  1%   +6.02% (p=0.002 n=6)
/read_body/TinyGo-12                      500.8n ± 1%   539.9n ±  1%   +7.82% (p=0.002 n=6)
/read_body_stream/TinyGo-12               504.7n ± 0%   508.4n ±  1%   +0.73% (p=0.011 n=6)
/read_body_stream_large/TinyGo-12         625.0n ± 1%   620.4n ±  1%   -0.74% (p=0.004 n=6)
/set_status_code/TinyGo-12                466.4n ± 1%   463.0n ±  1%   -0.72% (p=0.002 n=6)
/example_wasi/TinyGo-12                   7.291µ ± 1%   7.861µ ±  1%   +7.82% (p=0.002 n=6)
/get_header_names_large/TinyGo-12         1.442µ ± 6%   1.383µ ±  3%        ~ (p=0.145 n=6)
/get_header_values_exists/TinyGo-12       493.4n ± 1%   546.1n ±  0%  +10.66% (p=0.002 n=6)
/set_header_value_exists/TinyGo-12        619.1n ± 0%   700.6n ±  1%  +13.16% (p=0.002 n=6)
/set_header_value_not_exists/TinyGo-12    640.6n ± 1%   699.2n ±  1%   +9.16% (p=0.002 n=6)
/write_body/TinyGo-12                     481.8n ± 1%   516.6n ±  1%   +7.23% (p=0.002 n=6)
/set_uri/TinyGo-12                        593.6n ± 1%   614.0n ±  4%   +3.44% (p=0.002 n=6)
/add_header_value_exists/TinyGo-12        611.1n ± 0%   679.5n ±  1%  +11.18% (p=0.002 n=6)
/add_header_value_not_exists/TinyGo-12    634.4n ± 1%   714.1n ±  2%  +12.58% (p=0.002 n=6)
/remove_header_not_exists/TinyGo-12       542.2n ± 0%   586.5n ±  1%   +8.16% (p=0.002 n=6)
```

### CODE REVIEW

File: handler/body.go [hunk 1]
```diff
@@ -45,12 +45,12 @@ func (b wasmBody) Read(bytes []byte) (size uint32, eof bool) {
...
 	}
 
-	size, eof = read(b, uintptr(ptr), limit)
+	size, eof = read(b, ptr, limit)
 	runtime.KeepAlive(bytes) // keep bytes alive until ptr is no longer needed.
 	return
 }
 
-func read(b wasmBody, ptr uintptr, limit imports.BufLimit) (size uint32, eof bool) {
+func read(b wasmBody, ptr uint32, limit imports.BufLimit) (size uint32, eof bool) {
```
Comment (anuraaga) [inline handler/body.go:53]: `uint32` looks fine but just curious did something change in 0.28 requiring that?

**Review by anuraaga — APPROVED** [review]

File: handler/body.go (hunk 1 above)
Comment (codefromthecrypt) [reply handler/body.go:53]: `//go:wasmimport` is more restrictive on types accepted, particularly normal Go.

Comment (codefromthecrypt) [reply handler/body.go:53]: here's the tinygo one.. fishing for go.. https://github.com/tinygo-org/tinygo/blob/v0.28.1/compiler/symbol.go#L389-L393

Comment (anuraaga) [reply handler/body.go:53]: Thanks - no worries, glad I got to see it here before crashing into the wall :)

Comment (codefromthecrypt) [reply handler/body.go:53]: hmm not sure this is it as it seems to support unsafe.Pointer? https://github.com/golang/go/blob/ee46f0b5084461978432aa20df003ac52500b0f0/src/cmd/compile/internal/ssagen/abi.go#L369-L390

@achille-roussel do you know the canonical restricted list allowed for imports? I'd like to refer back to this issue if I am curious again later ;)

**Pull request merged**